src/index.ts

```diff
@@ -1,19 +1,944 @@
+
+    return ciphertextBuffer;
+  }
+
+  /**
+   * Decrypt data with the users JWK. This function assumes (and requires) a user is logged in and a valid encrypt() response.
+   * @param ciphertext The data to decrypt.
+   * @returns The decrypted data.
+   */
+  async decrypt(ciphertext: BinaryDataType): Promise<string> {
+    const { sub } = await this.ensureRequiredUserDataOrThrow();
+
+    const plaintext = await this.api.decrypt(ciphertext, sub);
+
+    return plaintext;
+  }
+
+  // SIGN:
+
+  // TODO: Add deprecation warning (and update all TSDocs according to what's on ArConnect and add references to their docs).
+
+  /**
+   * Generate a signature. This function assumes (and requires) a user is logged in.
+   * @param data The data to sign.
+   * @returns The {@linkcode Buffer} format of the signature.
+   */
+  async signature(data: string | BinaryDataType): Promise<Uint8Array> {
+    const { sub } = await this.ensureRequiredUserDataOrThrow();
+
+    const signatureBuffer = await this.api.sign(data, sub);
+
+    return signatureBuffer;
+  }
+
+  /**
+   * The signDataItem() function allows you to create and sign a data item object, compatible with arbundles. These data items can then be submitted to an ANS-104 compatible bundler.
+   * @param dataItem The data to sign.
+   * @returns The signed data item.
+   */
+  async signDataItem(dataItem: DataItem): Promise<ArrayBufferLike> {
+    const { sub, publicKey } = await this.ensureRequiredUserDataOrThrow();
+
+    const { data, tags, ...options } = dataItem;
+
+    const signer: Signer = {
+      publicKey: toBuffer(publicKey),
+      signatureType: 1,
+      signatureLength: 512,
+      ownerLength: 512,
+      sign: this.api.getSignerSignFn(sub),
+      // Note we don't provide `verify` as it's not used anyway:
+      // verify: () => true,
+    };
+
+    const opts: DataItemCreateOptions = {
+      ...options,
+      tags: this.addCommonTags(tags),
+    };
+
+    const dataItemInstance = createData(data, signer, opts);
+
+    // DataItem.sign() sets the DataItem's `id` property and returns its `rawId`:
+    await dataItemInstance.sign(signer);
+
+    // TODO: ArConnects types the return type as ArrayBuffer, but in the example that goes straight into the `DataItem`
+    // constructor, which only accepts Buffer, so this is not valid:
+    //     new DataItem(dataItemInstance.getRaw().buffer);
+    // Instead, we need to do:
+    //     new DataItem(Buffer.from(dataItemInstance.getRaw().buffer));
+    // Otherwise, if ArConnect's example is fine, the types are wrong and this function actually returns `Buffer`:
+    //     new DataItem(dataItemInstance.getRaw());
+    // In Othent's case, we prefer to return `ArrayBuffer`, as the return type from `dataItemInstance.getRaw()` cannot
+    // be handled bu the test repo's `replacer()` function so it doesn't seem to be a "proper" `Buffer` anyway.
+
+    return dataItemInstance.getRaw().buffer;
```

Comment:
@7i7o ArConnect specifies the return type of `signDataItem` as `ArrayBuffer`, so the last statement should be like this.

However, in their docs there's an example where they do:

````
// sign the data item
const signed = await window.arweaveWallet.signDataItem({
    data: "This is an example data",
    tags: [{
        name: "Content-Type",
        value: "text/plain"
    }]
});

// load the result into a DataItem instance
const dataItem = new DataItem(signed);
```

For that to work, the `return` should instead say:

   return dataItemInstance.getRaw();

Alternatively, their example should be updated to:

    const dataItem = new DataItem(Buffer.from(signed));